db/rdb/alpine.go

```diff
@@ -30,46 +31,69 @@ func (o *Alpine) InsertOval(root *models.Root, meta models.FetchMeta, driver *go
+		return tx.Rollback().Error
+	}
+	log15.Info("Refreshing...", "Family", root.Family, "Version", root.OSVersion)
+
 	old := models.Root{}
-	r := tx.Where(&models.Root{Family: root.Family, OSVersion: root.OSVersion}).First(&old)
-	if !r.RecordNotFound() {
+	r = tx.Where(&models.Root{Family: root.Family, OSVersion: root.OSVersion}).First(&old)
+	if r.Error != nil && !errors.Is(r.Error, gorm.ErrRecordNotFound) {
+		tx.Rollback()
+		return xerrors.Errorf("Failed to select old defs: %w", r.Error)
+	}
 
+	if r.RowsAffected > 0 {
+		// Delete data related to root passed in arg
 		defs := []models.Definition{}
-		tx.Model(&old).Related(&defs, "Definitions")
+		if err := tx.Model(&old).Association("Definitions").Find(&defs); err != nil {
+			tx.Rollback()
+			return xerrors.Errorf("Failed to select old defs: %w", err)
+		}
 		for _, def := range defs {
 			adv := models.Advisory{}
-			tx.Model(&def).Related(&adv, "Advisory")
-			if err := tx.Unscoped().Where("advisory_id = ?", adv.ID).Delete(&models.Cve{}).Error; err != nil {
-				tx.Rollback()
-				return fmt.Errorf("Failed to delete: %s", err)
-			}
-			if err := tx.Unscoped().Where("definition_id = ?", def.ID).Delete(&models.Advisory{}).Error; err != nil {
+			if err := tx.Model(&def).Association("Advisory").Find(&adv); err != nil {
 				tx.Rollback()
-				return fmt.Errorf("Failed to delete: %s", err)
+				return xerrors.Errorf("Failed to delete: %w", err)
 			}
-			if err := tx.Unscoped().Where("definition_id= ?", def.ID).Delete(&models.Package{}).Error; err != nil {
+			if err := tx.Select(clause.Associations).Unscoped().Where("id = ?", adv.ID).Delete(&adv).Error; err != nil {
 				tx.Rollback()
-				return fmt.Errorf("Failed to delete: %s", err)
+				return xerrors.Errorf("Failed to delete: %w", err)
 			}
-			if err := tx.Unscoped().Where("definition_id = ?", def.ID).Delete(&models.Reference{}).Error; err != nil {
+
+			if err := tx.Select(clause.Associations).Unscoped().Where("definition_id = ?", def.ID).Delete(&def).Error; err != nil {
 				tx.Rollback()
-				return fmt.Errorf("Failed to delete: %s", err)
+				return xerrors.Errorf("Failed to delete: %w", err)
 			}
 		}
 		if err := tx.Unscoped().Where("root_id = ?", old.ID).Delete(&models.Definition{}).Error; err != nil {
 			tx.Rollback()
-			return fmt.Errorf("Failed to delete: %s", err)
+			return xerrors.Errorf("Failed to delete: %w", err)
 		}
 		if err := tx.Unscoped().Where("id = ?", old.ID).Delete(&models.Root{}).Error; err != nil {
 			tx.Rollback()
-			return fmt.Errorf("Failed to delete: %s", err)
+			return xerrors.Errorf("Failed to delete: %w", err)
 		}
 	}
 
-	if err := tx.Create(&root).Error; err != nil {
+	if err := tx.Omit("Definitions").Create(&root).Error; err != nil {
 		tx.Rollback()
-		return fmt.Errorf("Failed to insert. cve: %s, err: %s",
-			pp.Sprintf("%v", root), err)
+		return xerrors.Errorf("Failed to insert. err: %w", err)
+	}
+
+	rootID := root.ID
+	if rootID == 0 {
+		rootID = 1
```

Comment:
What is the meaning of this code? Please comment on why you are assigning 1.

Reply:
There was a time when the rootID was set to 0 during the implementation, but when I checked it now, the ID was obtained correctly, so I fixed it.